src/deriver/queue_manager.py

```
@@ -117,37 +116,64 @@ async def cleanup(self) -> None:
+        return claimed_units
+
+    async def claim_work_unit(self, work_unit_key: str) -> bool:
+        async with tracked_db("claim_work_unit") as db:
+            try:
+                await db.execute(
+                    insert(models.ActiveQueueSession).values(
+                        work_unit_key=work_unit_key
+                    )
+                )
+                await db.commit()
+                logger.debug(f"Claimed work unit {work_unit_key}")
```

Ruff (0.12.2)

Logging statement uses f-string

(G004)